multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/TimeoutAsyncFlowableStep.java

```diff
@@ -50,4 +54,35 @@ private String getStepName() {
     }
 
     public abstract Duration getTimeout(ProcessContext context);
+
+    protected Integer extractUploadTimeoutFromAppAttributes(CloudApplicationExtended app, String supportedParameter) {
+        ApplicationAttributes appAttributes = ApplicationAttributes.fromApplication(app, app.getEnv());
+        Number taskExecutionTimeout = appAttributes.get(supportedParameter, Number.class);
+        return taskExecutionTimeout != null ? taskExecutionTimeout.intValue() : null;
+    }
+
+    protected void logTimeout(String message, Number timeout) {
+        getStepLogger().debug(message, timeout);
+    }
+
+    protected Duration getGivenTimeout(ProcessContext context, String supportedParameter, Variable<Duration> durationOperational,
```

Comment:
refactor name - 'getGivenTimeout' is too ambiguous - maybe change to 'calculateTimeout' or 'combineTimeoutValues'
'durationOperational' might be more clear as 'timeoutLocal' and 'durationGlobal' -> 'timeoutGlobal'
'supportedParameter' -> 'timeoutAppLevel' ?